CONTRIBUTING.md

```diff
@@ -1,341 +1,180 @@
-   * For MacOS, using Homebrew: `brew install kubernetes-helm`
-
-3. Start minikube.
-
-   For minikube version 0.26 and higher:
    ```bash
-   minikube start
+   git version
    ```
 
-   For older minikube versions:
-   ```bash
-   minikube start --extra-config=apiserver.Authorization.Mode=RBAC
-   ```
+1. Make a GitHub fork of [this
+   repository](https://github.com/jupyterhub/zero-to-jupyterhub-k8s) by creating
+   and then logging into your GitHub account and clicking the Fork button.
 
-   Note on troubleshooting: if you recently upgraded minikube and are now seeing
-   errors, you may need to clear out the `~/.minikube` and `~/.kube` directories
-   and reboot.
-
-4. Use the docker daemon inside minikube for building:
-   ```bash
-   eval $(minikube docker-env)
-   ```
+1. Clone your fork to your local computer.
 
-5. Clone the zero-to-jupyterhub repo:
    ```bash
-   git clone git@github.com:jupyterhub/zero-to-jupyterhub-k8s.git
+   git clone http://github.com/<YOUR-GITHUB-USERNAME>/zero-to-jupyterhub-k8s.git
    cd zero-to-jupyterhub-k8s
-   ```
 
-6. Create a virtualenv & install the libraries required for builds to happen:
-   ```bash
-   python3 -m venv .
-   source bin/activate
-   python3 -m pip install -r dev-requirements.txt
+   # make it easy to reference the projects GitHub repository as "upstream"
+   git remote add upstream https://github.com/jupyterhub/zero-to-jupyterhub-k8s
+
+   # make it obvious what you reference by renaming a reference to your
+   # personal GitHub repository to "fork"
+   git remote rename origin fork
    ```
 
-7. Now run `chartpress` to build the requisite docker images inside minikube:
-    ```bash
-    chartpress
-    ```
+### Prepare Virtual Machine software
 
-    This will build the docker images inside minikube & modify
-    `jupyterhub/values.yaml` with the appropriate values to make the chart
-    installable!
+A `Vagrantfile` is a way to prepare a Virtual Machine (VM), and we [have
+one](Vagrantfile) to prepare a VM for local development! We can use it to get
+a VM up and running, enter it with SSH, develop and run tests, and later shut
+down without influencing our system.
 
-8. Configure helm and minikube for RBAC:
-   ```bash
-   kubectl create clusterrolebinding add-on-cluster-admin --clusterrole=cluster-admin --serviceaccount=kube-system:default
-   kubectl --namespace kube-system create sa tiller
-   kubectl create clusterrolebinding tiller \
-       --clusterrole cluster-admin \
-       --serviceaccount=kube-system:tiller
-   helm init --service-account tiller
-   ```
+1. Install VirtualBox by [downloading and running an
+   installer](https://www.virtualbox.org/wiki/Downloads).
 
-9. Install / Upgrade JupyterHub Chart!
-   ```bash
-   helm upgrade --wait --install --namespace=hub hub jupyterhub/ -f minikube-config.yaml
-   ```
+1. Install Vagrant by [downloading and running an
+   installer](https://www.vagrantup.com/downloads.html).
```

Comment:
Oof. I thought part of the point of `kind` was that you didn't need to use a VM. Is that not true? There's an extremely high cost to using VMs, especially if you have to ssh into them to do any work.

Reply:
> FWIW, I can't get a working VM up with vagrant at the moment. I just get DNS failures and while it's a [common issue](https://www.vagrantup.com/docs/virtualbox/common-issues.html#dns-not-working) according to Vagrant, the linked instructions for fixing it don't actually work, so I can't test the current instructions. This doesn't appear to be a general virtualbox issue, because my vbox minikube VM has no issues.

Oh... I removed this snipped of code added by @manics two years ago in [xenial-setup.sh](xenial-setup.sh) that is a script running to initialize the VM.

```
# Workaround Minikube DNS problems
# https://github.com/kubernetes/minikube/issues/2027#issuecomment-338221646
cat << EOF > /etc/resolv.conf
nameserver 8.8.4.4
nameserver 8.8.8.8
EOF
sed -i -re "s/^(127.0.0.1\\s.+)/\\1 `hostname`/" /etc/hosts
```

By reading in the referenced issue, I see that this isn't a minikube specific problem. Does the inclusion of that in the `vagrant-vm-setup.sh` file make things work for you?